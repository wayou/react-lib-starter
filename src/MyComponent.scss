.my-component{
  // style rules go here
}
