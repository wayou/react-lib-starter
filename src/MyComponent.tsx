/*
 * @author: <user_name>
 * @date: <datetime>
 * @description: your component code
 */

import * as React from "react";
import "./MyComponent.scss";

interface IMyComponentProps {
  /**
   * custom class name
   */
  className?: string;

  /**
   * message to display
   */
  message?: string;
}

class MyComponent extends React.Component<IMyComponentProps, {}> {
  public static defaultProps: Partial<IMyComponentProps> = {
    className: "",
    message: "Allo!",
  };

  public render() {
    const { className, message } = this.props;
    return <div className={className}>{message}</div>;
  }
}

export default MyComponent;
