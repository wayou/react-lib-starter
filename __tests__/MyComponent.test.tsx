import * as React from "react";
import { create } from "react-test-renderer";
import MyComponent from "../src/MyComponent";

test("Component should rendered", () => {
  const component = create(<MyComponent message="Allo!" />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
