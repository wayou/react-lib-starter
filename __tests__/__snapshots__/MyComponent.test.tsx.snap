// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component should rendered 1`] = `
<div
  className=""
>
  Allo!
</div>
`;
